packages/nx/src/generators/tree.ts

```diff
@@ -357,20 +361,16 @@ export class FsTree implements Tree {
 
   private assertUnlocked() {
     if (this.locked) {
-      // TODO (v17): Remove condition
-      if (gt(nxVersion, '17.0.0')) {
-        throw new Error(
-          'The tree has already been committed to disk. It can no longer be modified. Do not modify the tree during a GeneratorCallback and ensure that Promises have resolved before the generator returns or resolves.'
-        );
-      } else {
-        output.warn({
-          title: 'Tree modified after commit to disk.',
-          bodyLines: [
-            'The tree has already been committed to disk. It can no longer be modified. Do not modify the tree during a GeneratorCallback and ensure that Promises have resolved before the generator returns or resolves.',
-            `This will be an error in version 16. Please open an issue on the Nx repo if experiencing this with a first-party plugin, or the plugin's repo if using a community plugin.`,
-          ],
-        });
-      }
+      output.error({
+        title: `File changes have already been written to disk. Further changes were attempted ${
+          this.logOperationId ? `while running ${this.logOperationId}.` : '.'
+        }`,
+        bodyLines: [
+          'It can no longer be modified. This commonly happens when a generator attempts to make further ',
+          'changes in its callback, or an asynchronous operation is still running after the generator completes.',
```

Comment:
Splitting a single sentence into multiple lines can make the sentence split weirdly in the terminal when the terminal is narrower than the parts. I think it's better not to break a sentence into various lines.

Reply:
Sure, that makes sense.